modules/javafx.graphics/src/main/java/javafx/scene/Scene.java

```diff
@@ -1346,14 +1347,13 @@ static WritableImage doSnapshot(Scene scene,
             context.camera = null;
         }
 
-        // Grab the lights from the scene
-        context.lights = null;
-        if (scene != null && !scene.lights.isEmpty()) {
-            context.lights = new NGLightBase[scene.lights.size()];
-            for (int i = 0; i < scene.lights.size(); i++) {
-                context.lights[i] = scene.lights.get(i).getPeer();
-            }
-        }
+        // Grab the lights from the scene and/or subscene
+        Stream<NGLightBase> lights = Stream.concat(
+            Optional.ofNullable(scene).stream().flatMap(s -> s.lights.stream()).map(LightBase::getPeer),
+            Optional.ofNullable(subScene).stream().flatMap(s -> s.getLights().stream()).map(LightBase::getPeer)
```

Comment:
For a node in a SubScene this will apply the lights from both the Scene and SubScene, which is not the right behavior. It should apply the lights from `scene` only if `subScene` is null.